### DISCUSSION TITLE
SSO & Parameter Store

### DISCUSSION THREAD

**Discussion by johanneskvamme** [awspring/spring-cloud-aws#1131] [Q&A] [answered]
Hello! 👋 

I am using currently migrating from a manual setup to using spring-cloud-aws-starter-parameter-store:3.1.0 for variables.
However, I use AWS SSO for authentication towards AWS for most things. Is there a way to have this dependency get the credentials from `.aws/sso/cache` too? 

It does not seem to be able to autoconfigure with SSO, all I get is this: 

```
***************************
APPLICATION FAILED TO START
***************************

Description:

Config data resource '[ParameterStoreConfigDataResource@2d35442b context = '/config/app', optional = false]' via location 'aws-parameterstore:/config/app' does not exist

Action:

Check that the value 'aws-parameterstore:/config/app' at class path resource [application-local.yml] - 8:9 is correct, or prefix it with 'optional:'


Process finished with exit code 1
```

**Comment by maciejwalkowiak** [comment]
I'm not very familiar with this topic. How did you get credentials from this file before? If you have custom credentials provider you can configure it in bootstrap context like explained in docs: https://docs.awspring.io/spring-cloud-aws/docs/3.1.1/reference/html/index.html#customizing-ssmclient

Does it help?

**Reply by johanneskvamme** [reply]
Thank you for replying.

> How did you get credentials from this file before?   

Earlier, I used the AWS CLI to set all environment properties before boot. The AWS CLI automatically picks up the SSO credentials. Now I am moving to using `spring-cloud-aws-starter-parameter-store` to set them instead, and this does not automatically pick up that I am signed in.

> If you have custom credentials provider you can configure it in bootstrap context like explained in docs. 

I tried setting this up with the new [(Spring Boot 3) META-INF](https://stackoverflow.com/questions/75044304/custom-spring-boot-3-starter-does-not-create-configurationproperties-beans) setup, but I am not sure whether it detects the file or not. 
Setting it in the old `META-INF/spring.factories` did not seem to do anything either. So maybe there is something else with my setup.

Even adding it with `this.addBootstrapRegistryInitializer(ParameterStoreBootstrapConfiguration())` in Application did nothing, as far as I can tell

**Comment by christoph-herb-mw** [comment]
i have the same issue. Did you find a solution for this problem?

**Reply by johanneskvamme** [reply]
Sadly, no.
I ended up using a work-around through Jetbrains' Intellij, where the AWS plugin supports SSO login and worked with the automatic credential configuration for this dependency. Since I only used this plugin for local development, that was enough for me to keep working.

**Reply by christoph-herb-mw** [reply, marked as answer]
I found a solution for this problem. If you add the following dependencies to your project, it works just fine:

```gradle
  implementation 'io.awspring.cloud:spring-cloud-aws-starter-parameter-store:3.1.1'
  implementation 'software.amazon.awssdk:sso:2.26.5'
  implementation 'software.amazon.awssdk:ssooidc:2.26.5'
```

**Reply by MatejNedic** [reply]
@FluidSense @christoph-herb-mw 
Yes, you need to include sso dependency in other for AWS `DefualCredentialsProvider` to include it.

No implementation needed since AWS SDK already takes care of it but you need dependency included.
